src/script/user/UserPermission.ts

```diff
@@ -19,45 +19,40 @@
 
 import {capitalizeFirstChar} from 'Util/StringUtil';
 
-let bitsCounter = 0;
-
-/* eslint-disable sort-keys */
-/**
- * Enum for different team permissions.
- */
-const TEAM_FEATURES = {
-  NONE: 0,
-  CREATE_CONVERSATION: 1 << bitsCounter++,
-  DELETE_CONVERSATION: 1 << bitsCounter++,
-  ADD_TEAM_MEMBER: 1 << bitsCounter++,
-  REMOVE_TEAM_MEMBER: 1 << bitsCounter++,
-  ADD_CONVERSATION_MEMBER: 1 << bitsCounter++,
-  REMOVE_CONVERSATION_MEMBER: 1 << bitsCounter++,
-  GET_BILLING: 1 << bitsCounter++,
-  SET_BILLING: 1 << bitsCounter++,
-  SET_TEAM_DATA: 1 << bitsCounter++,
-  GET_MEMBER_PERMISSIONS: 1 << bitsCounter++,
-  GET_TEAM_CONVERSATIONS: 1 << bitsCounter++,
-  DELETE_TEAM: 1 << bitsCounter++,
-  SET_MEMBER_PERMISSIONS: 1 << bitsCounter++,
+// tslint:disable:object-literal-sort-keys
+enum TEAM_FEATURES {
+  NONE = 0,
+  CREATE_CONVERSATION = 1 << 0,
+  DELETE_CONVERSATION = 1 << 1,
+  ADD_TEAM_MEMBER = 1 << 2,
+  REMOVE_TEAM_MEMBER = 1 << 3,
+  ADD_CONVERSATION_MEMBER = 1 << 4,
+  REMOVE_CONVERSATION_MEMBER = 1 << 5,
+  GET_BILLING = 1 << 6,
+  SET_BILLING = 1 << 7,
+  SET_TEAM_DATA = 1 << 8,
+  GET_MEMBER_PERMISSIONS = 1 << 9,
+  GET_TEAM_CONVERSATIONS = 1 << 10,
+  DELETE_TEAM = 1 << 11,
+  SET_MEMBER_PERMISSIONS = 1 << 12,
 };
 
-const PUBLIC_FEATURES = {
-  CREATE_GROUP_CONVERSATION: 1 << bitsCounter++,
-  CREATE_GUEST_ROOM: 1 << bitsCounter++,
-  UPDATE_CONVERSATION_SETTINGS: 1 << bitsCounter++,
-  UPDATE_GROUP_PARTICIPANTS: 1 << bitsCounter++,
-  MANAGE_SERVICES: 1 << bitsCounter++,
-  MANAGE_TEAM: 1 << bitsCounter++,
-  INVITE_TEAM_MEMBERS: 1 << bitsCounter++,
-  CHAT_WITH_SERVICES: 1 << bitsCounter++,
-  SEARCH_UNCONNECTED_USERS: 1 << bitsCounter++,
+enum PUBLIC_FEATURES {
+  CREATE_GROUP_CONVERSATION = 1 << 13,
+  CREATE_GUEST_ROOM = 1 << 14,
+  UPDATE_CONVERSATION_SETTINGS = 1 << 15,
+  UPDATE_GROUP_PARTICIPANTS = 1 << 16,
+  MANAGE_SERVICES = 1 << 17,
+  MANAGE_TEAM = 1 << 18,
+  INVITE_TEAM_MEMBERS = 1 << 19,
+  CHAT_WITH_SERVICES = 1 << 20,
+  SEARCH_UNCONNECTED_USERS = 1 << 21,
```

Comment:
It's way better readable like this.

Reply:
the other option had some weird beauty.. can't decide :)

Reply:
and no magic numbers with the old approach

Reply:
These numbers are not magic, they are just counting up. The old approach was using a changing variable to generate a static, which is dangerous IMO.

Reply:
Now you can even sort the object and will not destroy it's functionality.

Reply:
Please don't sort this one!! It's gonna be awful and very error prone to add new values, else!!!

Reply:
I think we took the decision to use a variable with @AndyLnd because it leaves less room for error from a human. You can just copy paste a line and create the new entry without worrying about the current state of the enum

Reply:
> You can just copy paste a line and create the new entry without worrying about the current state of the enum

But you can easily paste it into the wrong line without knowing directly what the resulting number will be since you need to count `bitsCounter`. And you also have to take into consideration that `bitsCounter` is incrementing **after** it is written into the object...

Reply:
The thing is you never (or very rarely?) need to know what is the value is.
It is true that you can copy at the wrong place, I am open to using fixed values.

Just the sorting part, I really think it should not be done.

Reply:
> Just the sorting part, I really think it should not be done.

I agree.